src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,301 @@
+            } else {
+                task.markAsDone();
+                System.out.println("We did it! Good job little guy.");
+                saveToDisk();
+            }
+            System.out.println("\t" + task);
+        } catch (NumberFormatException nfe) {
+            System.out.println("Oh, no! I don't get what you are trying to say!");
+        } catch (IndexOutOfBoundsException ioobe) {
+            System.out.println("Oh, no! This task does not exist!");
+        }
+    }
+
+    /**
+     * Prints the entire list of tasks
+     */
+    private static void printList() {
+        int doneCount = 0;
+        System.out.println("Here we go! These are the tasks you have:");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            int order = i + 1;
+            if (task.isDone()) {
+                doneCount++;
+            }
+            System.out.printf("\t%d. %s%n", order, task);
+        }
+        System.out.printf("You have %d tasks and you completed %d of them%n", tasks.size(), doneCount);
+    }
+
+    /**
+     * Save the current list of tasks to the file "data/duke.txt" on the disk
+     */
+    private static void saveToDisk() {
+        try {
+            //Check if directory exists if not create it
+            checkDataDir();
+            FileWriter fw = new FileWriter("./data/duke.txt");
+            for (Task task : tasks) {
+                String rawData = task.getRawData() + System.lineSeparator();
+                fw.append(rawData);
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Fail to save data to disk");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Load data from a file "data/duke.txt" on the disk
+     */
+    private static void loadFromDisk() {
+        try {
+            File f = new File("./data/duke.txt");
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                Task task = Task.decodeTask(line);
+                if (task != null) { //If this task is not invalid
+                    tasks.add(task);
+                }
+            }
+        } catch (FileNotFoundException fnfe) {
+            System.out.println("Data file not found");
+        }
+    }
+
+    /**
+     * Check if data/ directory exists, if it does not exist, create it
+     *
+     * @throws IOException Throws if there are any IO exception
+     */
+    private static void checkDataDir() throws IOException {
+        Path path = Paths.get("./data");
+        if (!Files.exists(path)) {
+            Files.createDirectory(path);
+        }
+    }
+
+    /**
+     * Adds new task to the list
+     *
+     * @param type Type of the new task
+     * @param args Arguments that are passed with the type
+     */
+    public static void addTask(Task.Type type, String args) {
```

Comment:
consider handling this in the task class

Reply:
Makes sense. So I suppose it can be static in task class right?